nix/pkgs/haskell/haskell.nix

```diff
@@ -57,9 +56,93 @@ let
       "https://github.com/input-output-hk/Win32-network"."94153b676617f8f33abe8d8182c37377d2784bd1" = "0pb7bg0936fldaa5r08nqbxvi2g8pcy4w3c7kdcg7pdgmimr30ss";
       "https://github.com/input-output-hk/hedgehog-extras"."8bcd3c9dc22cc44f9fcfe161f4638a384fc7a187" = "12viwpahjdfvlqpnzdgjp40nw31rvyznnab1hml9afpaxd6ixh70";
     };
+    cabalProjectLocal = lib.optionalString pkgs.stdenv.hostPlatform.isWindows ''
+      -- When cross compiling we don't have a `ghc` package
+      package plutus-tx-plugin
+        flags: +use-ghc-stub
+
+      -- The following is needed because Nix is doing something crazy.
```

Comment:
Do we need these?

Reply:
Without `use-ghc-stub` I think the solver will fail while trying to find a `ghc` package.  I'll double check and update the comment to reflect why they are needed.

Reply:
Still needs a better comment

Reply:
bump

Reply:
Just two of these were needed and I have added a comment explaining that it is the `doctest` dependency that causes issues.